### DISCUSSION TITLE
GetCurrentTransaction().IsActive?

### DISCUSSION THREAD

**Discussion by nfplee** [nhibernate/nhibernate-core#2522] [General] [answered]
I've just upgrade NHibernate it is telling me session.Transaction is now obselete and to use GetCurrentTransaction instead. However I noticed in the [ThreadLocalSessionContext.cs](/nhibernate/nhibernate-core/blob/c60136b69b205e5487e29db5c8f52d2405842b7e/src/NHibernate/Context/ThreadLocalSessionContext.cs#L82) file that it says:

```
var transaction = orphan.GetCurrentTransaction();
if (transaction?.IsActive == true)
```

Is the IsActive check needed? Couldn't it just check that transaction is not null since I thought GetCurrentTransaction would return null if it's not active.

**Comment by bahusoid** [comment, marked as answer]
> Is the IsActive check needed?  I thought GetCurrentTransaction would return null if it's not active.

Yeah it's not needed.  You can skip this check in your code. I don't remember why it's been left but for some  legacy obscure custom transaction handling cases

**Comment by hazzik** [comment]
Probably we don't need it either.

**Comment by nfplee** [comment]
Thanks both of you.
